[NC-374] [NC-371] Native maps issues

wegiswes opened the pull request:
## Checklist
- Contains unit tests ❌
- Contains breaking changes ❌
- Contains Atlas changes ❌
- Compatible with: MX 9️⃣

#### Native specific
- Works in Android ✅
- Works in iOS ✅
- Works in Tablet ✅

## This PR contains
- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Other (describe)

## What is the purpose of this PR?
To fix an issue where where latitude or longtitude with a value of 0 gave unexpected results.
To fix an issue where dynamic markers added on the fly were not added to the maps.

## Relevant changes
We now check if latitude and longitude are typeof number.
We added a check in componentDidUpdate where we compare the amount of markers coming from props with the amount of markers in the current state.

## What should be covered while testing?
If the mentioned issues are fixed.

Commit:
fix(maps-native): 0 value seen as false

Commit:
fix(maps-native): adding dynamic markers on the fly

Commit:
fix(maps-native): add changelog

Commit:
fix(maps-native): add changelog

pratikgoyal9 commented:
Tested the below bugs:
1. To fix an issue where latitude or longitude with a value of 0 gave unexpected results - **Closed**
2. To fix an issue where dynamic markers added on the fly were not added to the maps. - **Reopen**

Commit:
fix(maps-native): fix coordinate validation

Commit:
fix(maps-native): add minus to coord validation check

jordanch commented on packages/pluggableWidgets/maps-native/src/Maps.tsx:
```diff
@@ -245,6 +249,10 @@ export class Maps extends Component<Props, State> {
     }
 }
 
+function isValidCoordinate(value: Big | number | undefined): boolean {
```
If you'd like, you could import `Option` from `mendix` module, then write like `Option<Big | numer>`. If you don't want to change it, it's good to know for next time :)

jordanch reviewed (dismissed):
LGTM 👍 left one optional comment

Commit:
fix(maps-native): update typings

Commit:
fix(maps-native): onclick event markers

Commit:
fix(maps-native): changing markers not triggering update

fix(maps-native): changing markers not triggering update

fix(maps-native): changing markers not triggering update

jordanch reviewed (dismissed):
LGTM 👍 Do you want to create a JIRA for those TODO items?

diego-antonelli reviewed (dismissed):
Approving for Web Content based on the `.gitignore` changes.

wegiswes committed:
fix(maps-native): Improve parseCoordinate error message

fix(maps-native): Improve parseCoordinate error message
